bridge/shared/interfaces/secret.ts

```diff
@@ -5,9 +5,15 @@ export interface SecretKeyValuePair {
   value: string;
 }
 
-export interface ISecret {
+interface ISecret {
   name: string;
   scope: SecretScope;
-  data?: SecretKeyValuePair[];
+}
+
+export interface IClientSecret extends ISecret {
   keys?: string[];
```

Comment:
We don't need the `?` here anymore.
Before it was either `kays: string[]` or `data: SecretKeyValuePair`. It was neither both nor none